packages/core/src/api/user-manager.ts

```diff
@@ -460,6 +462,58 @@ class UserManager {
     }
   }
 
+  async getInboxKeys() {
+    if (this.cachedInboxKeys) return this.cachedInboxKeys;
+    try {
+      let user = await this.getUser();
+      if (!user) return;
+
+      if (!user.inboxKeys) {
+        const token = await this.tokenManager.getAccessToken();
+        user = await http.get(`${constants.API_HOST}${ENDPOINTS.user}`, token);
+      }
+      if (!user) return;
+
+      const userEncryptionKey = await this.getEncryptionKey();
+      if (!userEncryptionKey) return;
+
+      if (!user.inboxKeys) {
+        const keys = await this.db.crypto().generateRandomKeyPair();
+        user.inboxKeys = {
+          public: keys.publicKey,
+          private: await this.db
+            .storage()
+            .encrypt(userEncryptionKey, JSON.stringify(keys.privateKey))
+        };
+        await this.updateUser({ inboxKeys: user.inboxKeys });
+        return keys;
+      }
+
+      const plainData: SerializedKeyPair = {
+        publicKey: user.inboxKeys.public,
+        privateKey: JSON.parse(
+          await this.db
+            .storage()
+            .decrypt(userEncryptionKey, user.inboxKeys.private)
+        )
+      };
+      this.cachedInboxKeys = plainData;
+      return this.cachedInboxKeys;
+    } catch (e) {
+      logger.error(e, "Could not get inbox encryption keys.");
+      if (e instanceof Error)
+        throw new Error(
+          `Could not get inbox encryption keys. Error: ${e.message}`
+        );
+    }
+  }
```

Comment:
We now have 3 methods that are more of the same. Let's refactor them so they share the common bits.